docs/libcurl/opts/CURLOPT_HTTP_TRAILER_FUNCTION.3

```diff
@@ -0,0 +1,110 @@
+.\" *
+.\" * Copyright (C) 1998 - 2017, Daniel Stenberg, <daniel@haxx.se>, et al.
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.haxx.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLOPT_HTTP_TRAILER_FUNCTION 3 "14 Aug 2018" "libcurl 7.63.0" "curl_easy_setopt options"
+
+.SH NAME:
+CURLOPT_HTTP_TRAILER_FUNCTION \- Callback to fill trailing headers list to be sent in a chunked transfer.
+		
+.SH SYNOPSIS:
+#include <curl.h>
+
+void curl_trailer_callback(struct curl_slist ** list, void *userdata);
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_HTTP_TRAILER_FUNCTION, curl_trailer_callback *func);
+
+.SH DESCRIPTION:
+Pass a pointer to a callback function.
+
+This callback function will be called once right before sending the final
+CR LF in an HTTP chunked transfer to fill a list of trailing headers to be
+sent before finishing the HTTP transfer.
+
+You can set the userdata argument with the CURLOPT_HTTP_TRAILER_DATA option.
+
+The trailing headers included in the linked list must not be CRLF-terminated,
+because libcurl will add the appropriate line termination characters after
+each header item.
+
+If you use curl_slist_append to add trailing headers to the curl_slist then
+libcurl will duplicate the strings, and will free the curl_slist and the
+duplicates once the trailers have been sent.
+
+If one of the trailing headers is not formatted correctly
+(i.e. HeaderName: headerdata) it will be ignored and an info message
+will be emitted.
+
+If you set this option to NULL, then the HTTP transfer proceeds as usual
+without any interruptions.
+
+.SH DEFAULT:
+NULL
+
+.SH PROTOCOLS:
+HTTP
+
+.SH EXAMPLE:
+#include <curl/curl.h>
+
+static void trailer_cb(struct curl_slist **tr, void *data)
+{
+    /* libcurl will free the list as soon as it becomes not needed */
+	*tr = curl_slist_append(*tr, "My-super-awesome-trailer: trailer-stuff");
+	(void)data;
+}
+
+CURL *curl = curl_easy_init();
+if(curl) {
+
+    /* Set the URL of the request */
+    curl_easy_setopt(curl, CURLOPT_URL, "http://example.com/");
+    /* Now set it as a put */
+    curl_easy_setopt(curl, CURLOPT_PUT, 1L);
+    
+    /* Assuming we have a function that will return the data to be pushed 
+       Let that function be read_cb */
+    curl_easy_setopt(curl, CURLOPT_READFUNCTION, read_cb);
+
+    /* We need to announce the trailers to come 
+       See CURLOPT_READFUNCTION for more on the matter */
```

Comment:
But this isn't explained there, is it?

Reply:
This is a weird combined leftover from a previous version of the patch and a copy pasted example that was modified. It's not required at all. I'm really sorry for this inconsistency.